js/multilingual_survey_settings.js

```diff
@@ -550,22 +633,59 @@ Multilingual.loadSurveySettings = function() {
 			}
 		}
 	})
+	
+	// update hide_lang checkbox
+	if (collections.basic_settings.hide_lang == '1') {
+		$("#hide_lang").prop('checked', true)
+	} else {
+		$("#hide_lang").prop('checked', false)
+	}
+	
 }
 
 Multilingual.getSettings();
 
+Multilingual.translateSettings = function() {
+	// get selected language from #ml-translate-language
+	var language_iso_code = $("select#ml-translate-language").val();
+	if (!language_iso_code) {
+		return;
+	}
+	
+	$.ajax({
+		url: Multilingual.ajax_url + '&translate_settings_iso_code=' + language_iso_code,
+		complete: function(response) {
+			Multilingual.response = response;
+			var translations = response.responseJSON.translations;
+			if (response.error) {
+				alert(response.error);
+			} else if (!Array.isArray(translations)) {
+				alert("The Multilingual module wasn't able to retrieve a valid translations table!");
+			} else if (translations.length < 1) {
+				alert("The Multilingual module was able to retrieve translations, but the table is empty.");
+			} else {
+				for (var i = 0; i < translations.length; i++) {
+					$('.ml-text-setting:eq(' + i + ')').val(translations[i]);
+				}
+			}
+		}
+	});
+}
+
 $( document ).ready(function() {
 	Multilingual.getLanguages();
+	Multilingual.instructions_mce = tinyMCE.get('mce_0');
+	Multilingual.acknowledgement_mce = tinyMCE.get('mce_2');
 	
 	// disable text settings when both tinyMCE editors are init'ed
-	tinyMCE.editors["instructions"].on('init', function(e) {
+	Multilingual.instructions_mce.on('init', function(e) {
 		Multilingual.instructionsEditorReady = true;
 		if (Multilingual.acknowledgementEditorReady && Multilingual.instructionsEditorReady) {
 			Multilingual.disableTextSettings();
 			Multilingual.getSurveySettings();
 		}
 	})
-	tinyMCE.editors["acknowledgement"].on('init', function(e) {
+	Multilingual.acknowledgement_mce.on('init', function(e) {
```

Comment:
Fetching tinymce text is only done on the Survey Settings. I'm highly confident referencing them in this way will not introduce regressions that affect functionality that existed before. But I'm not confident that this will continue to work. If you know of any place where the TinyMCE editors are documented for the Survey Settings page in REDCap, please forward this information to me. I recall being able to reference editor values the old way, then suddenly they did not, and I had to search to find a way to reference them, and I found using the tinyMCE.get method against id's "mce_0" and "mce_2" to work -- but again I'm not confident this will continue to work.

Reply:
While they are unspecific, I think the likelihood of Rob changing those `mce_#` IDs changing is actually pretty low.  With that said, you could do something like this if you wanted to reference them by `name` instead:

```
var instructionsId = $('textarea[name=instructions]').attr('id')
Multilingual.instructions_mce = tinyMCE.get(instructionsId)
```

On a semi-related note, if there is a possibility/concern that tinyMCE objects could get replaced/reinitialized on the fly after certain actions, you could replace `Multilingual.instructions_mce` with a function to always get the current refernce on the fly instead of caching it:
```
Multilingual.getInstructionsMCE = function(){
    var instructionsId = $('textarea[name=instructions]').attr('id')
    return tinyMCE.get(instructionsId)
}
```